src/Microsoft.Azure.CosmosRepository/Specification/BaseSpecification.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+namespace Microsoft.Azure.CosmosRepository.Specification
+{
+    /// <inheritdoc cref="ISpecification{T}"/>
+    public abstract class BaseSpecification<T> : ISpecification<T>
+        where T : IItem
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        protected virtual ISpecificationBuilder<T> Query { get; }
+        /// <summary>
+        /// Initialize specificaiton and add filters later
+        /// </summary>
+        public BaseSpecification()
+        {
+            Query = new SpecificationBuilder<T>(this);
+
+        }
+        /// <summary>
+        /// Initialize specificaiton with a list of filter criteria
+        /// </summary>
+        /// <param name="filters"></param>
+        public BaseSpecification(Expression<Func<T, bool>>[] filters)
+        {
+            Query = new SpecificationBuilder<T>(this);
+            foreach(Expression<Func<T, bool>> filter in filters)
+            {
+                Query.Where(filter);
+            }
+        }
+        /// <inheritdoc/>
+        public IEnumerable<WhereExpressionInfo<T>> WhereExpressions { get; } = new List<WhereExpressionInfo<T>>();
```

Comment:
Would these want to be `IReadOnlyList<T>` ? Do we want people to edit these collections directly or use the builder?

Reply:
I think all work should be done via the builders